toothpick-compiler/src/test/java/toothpick/compiler/memberinjector/ObfuscationFriendlyMemberInjectorRegistryTest.java

```diff
@@ -0,0 +1,151 @@
+package toothpick.compiler.memberinjector;
+
+import com.google.common.base.Joiner;
+import com.google.testing.compile.JavaFileObjects;
+
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+
+import javax.tools.JavaFileObject;
+
+import static com.google.common.truth.Truth.assert_;
+import static com.google.testing.compile.JavaSourceSubjectFactory.javaSource;
+
+public class ObfuscationFriendlyMemberInjectorRegistryTest {
+  @Test
+  public void testASimpleRegistry() {
+    JavaFileObject source = JavaFileObjects.forSourceString("test.TestASimpleRegistry", Joiner.on('\n').join(//
+        "package test;", //
+        "import javax.inject.Inject;", //
+        "public class TestASimpleRegistry {", //
+        "  @Inject String s; ", //
+        "}" //
+    ));
+
+    JavaFileObject expectedSource = JavaFileObjects.forSourceString("toothpick/MemberInjectorRegistry", Joiner.on('\n').join(//
+            "package toothpick;",
+                    "",
+                    "import java.lang.Class;",
+                    "import java.lang.Override;",
+                    "import java.lang.String;",
+                    "import java.util.HashMap;",
+                    "import java.util.Map;",
+                    "import toothpick.registries.memberinjector.AbstractMemberInjectorRegistry;",
+                    "",
+                    "public final class MemberInjectorRegistry extends AbstractMemberInjectorRegistry {",
+                    "    private final Map<String, MemberInjector> factories = new HashMap<>();",
+                    "",
+                    "    public MemberInjectorRegistry() {",
+                    "      factories.put(\"test.TestASimpleRegistry\", new test.TestASimpleRegistry$$MemberInjector());",
```

Comment:
This is a concern: it means that we instantiate all the injectors or the factories at startup when the registry itself is instantiated. This is probably an issue: it's gonna be very costly to use registries this way and startup time is probably gonna get much slower because of that.